### ISSUE TITLE
Merge overlapping times with 24:00 as range.

### ISSUE THREAD

**Issue by MathieuMaas** [spatie/opening-hours#187]
Hi, 
I am not sure if this is supposed to work like this but we had one of our users use the 24:00 midnight time which causes to crash on overlapping time ranges.(Time ranges 00:00-00:45 and 05:00-00:00 overlap).
I could fix it by having 23:59 as end time but this is not exactly what I expected to happen.
Is this something that should be possible?

Example of the issue:
```
OpeningHours::createAndMergeOverlappingRanges([
     'monday' => ['00:00-00:45', '05:00-24:00'],
])
```

Did some extra testing those cases do work:
```
OpeningHours::createAndMergeOverlappingRanges([
      'monday' => ['00:00-00:45', '05:00-00:00'],
]);
```

```
OpeningHours::createAndMergeOverlappingRanges([
      'monday' => ['24:00-00:45', '05:00-24:00'],
]);
```

**Comment by kylekatarnls** [comment]
Hi,

On a single day (here Monday), those 2 ranges do not overlap, so I don't see what you expect as an output. Also those code do not crash (tried on http://try-carbon.herokuapp.com/)

Please provide a reproducible case (minimum code chunk I can run in http://try-carbon.herokuapp.com/ to reproduce the crash/bug)

Precise what error you exactly get so I can check I got the same, then detail what result you were expecting from this given code.

Thanks.

**Comment by MathieuMaas** [comment]
Issue can be closed, seems to be working in the 2.0 versions
